mmengine/utils/package_utils.py

```diff
@@ -45,15 +43,31 @@ def get_installed_path(package: str) -> str:
     """
     import importlib.util
 
-    from pkg_resources import DistributionNotFound, get_distribution
-
     # if the package name is not the same as module name, module name should be
     # inferred. For example, mmcv-full is the package name, but mmcv is module
     # name. If we want to get the installed path of mmcv-full, we should concat
     # the pkg.location and module name
     try:
-        pkg = get_distribution(package)
-    except DistributionNotFound as e:
+        dist = distribution(package)
+        # In importlib.metadata, we use dist.locate_file() or files
+        if hasattr(dist, 'locate_file'):
+            # Python 3.9+
+            # locate_file returns PathLike, need to access parent
+            locate_result: Any = dist.locate_file('')
+            location = str(locate_result.parent)
+        elif hasattr(dist, '_path'):
+            # Python 3.8 - _path is a pathlib.Path object
+            # We know _path exists because we checked with hasattr
+            dist_any: Any = dist
+            location = str(dist_any._path.parent)  # type: ignore[attr-defined]
+        else:
+            # Fallback: try to find via importlib
+            spec = importlib.util.find_spec(package)
```

Comment:
Duplicated with line 73. It's better to keep the logic of `distribution package` and `import package` independent